include/vara/Feature/OrderedFeatureVector.h

```diff
@@ -13,40 +13,51 @@ namespace vara::feature {
 /// Feature::operator<.
 class OrderedFeatureVector {
 public:
-  using ordered_feature_iterator = typename std::vector<Feature *>::iterator;
+  using ordered_feature_iterator =
+      typename llvm::SmallVector<Feature *, 3>::iterator;
   using const_ordered_feature_iterator =
-      typename std::vector<Feature *>::const_iterator;
+      typename llvm::SmallVector<Feature *, 3>::const_iterator;
 
   OrderedFeatureVector() = default;
+  OrderedFeatureVector(std::initializer_list<Feature *> Init) { insert(Init); }
   template <class FeatureIterTy>
-  OrderedFeatureVector(FeatureIterTy Start, FeatureIterTy End) {
-    insert(llvm::iterator_range(std::move(Start), std::move(End)));
+  OrderedFeatureVector(FeatureIterTy Begin, FeatureIterTy End) {
+    insert(std::move(Begin), std::move(End));
```

Comment:
```suggestion
    insert(std::forward<FeatureIterTy>(Begin), std::forward<FeatureIterTy>(End));
```
You need to use forward here because you don't not the type `FeatureIterTy`